app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,177 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework import status
+
+from core.models import Recipe, Ingredient
+
+from recipe.serializers import RecipeDetailSerializer
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def specific_recipe_url(recipe_id):
+    """Return recipe detail URL"""
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def sample_ingredient(recipe, name='Sample ingredient'):
+    """Create and return a sample ingredient"""
+    return Ingredient.objects.create(recipe=recipe, name=name)
+
+
+def sample_recipe(**params):
+    """Create and return a sample recipe"""
+    defaults = {
+        'name': 'Sample recipe',
+        'description': 'Sample recipe description'
+    }
+    defaults.update(params)
+
+    return Recipe.objects.create(**defaults)
+
+
+class RecipeApiTests(TestCase):
+
+    def test_retrieve_recipes(self):
+        """Test retrieving a list of recipes"""
+        recipe1 = sample_recipe(name='Steak pie')
+        sample_recipe(name='Pizza')
+
+        sample_ingredient(recipe1)
+
+        res = self.client.get(RECIPES_URL)
+
+        recipes = Recipe.objects.all().order_by('name')
+        serializer = RecipeDetailSerializer(recipes, many=True)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_filter_recipes_by_name(self):
+        """Test returning a list of recipes filtered by name"""
+        recipe1 = sample_recipe(name='Pizza Margarita')
+        recipe2 = sample_recipe(name='Paella')
+        recipe3 = sample_recipe(name='Pizza Carbonara')
+        recipe4 = sample_recipe()
+
+        sample_ingredient(recipe1)
+
+        res = self.client.get(RECIPES_URL, {'name': 'Pizz'})
+
+        serializer1 = RecipeDetailSerializer(recipe1)
+        serializer2 = RecipeDetailSerializer(recipe2)
+        serializer3 = RecipeDetailSerializer(recipe3)
+        serializer4 = RecipeDetailSerializer(recipe4)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer1.data, res.data)
+        self.assertIn(serializer3.data, res.data)
+        self.assertNotIn(serializer2.data, res.data)
+        self.assertNotIn(serializer4.data, res.data)
```

Comment:
similarly to the comment above, we could check the data we get is what we expect and have the serializer tests separate:

```
assert len(res.data) == 2
assert {'Pizza Margarita', 'Pizza Carbonara'} in {recipe['name'] for recipe in res.data}
```

Reply:
Agreed! Adding a new field shouldn't break the test.
Is it fine if I leave it like that on this exercise?